spec/components/rails_bootstrap/badge_component_spec.rb

```diff
@@ -0,0 +1,52 @@
+require "rails_helper"
+
+RSpec.describe RailsBootstrap::BadgeComponent, type: :component do 
+  context "variation attribute tests" do
+    it "renders default variation" do 
+      subject = render_inline(RailsBootstrap::BadgeComponent.new()) { "This is a default badge" }
+
+      expect(subject.css("span[class='badge badge-primary']").to_html).not_to be_empty
+    end
+
+    it "renders secondary style" do 
+      subject = render_inline(RailsBootstrap::BadgeComponent.new(variation: "secondary")) { "This is the secondary style badge" }
+
+      expect(subject.css("span[class='badge badge-secondary']").to_html).not_to be_empty
+    end
+
+    it "renders info style" do 
```

Comment:
Since we already tested that setting `variation` to secondary works as expected we don't really need to show it for all the possible values of `variation` . It is reluctant and thus we can remove the test for `info` and `success`